src/mescal/agave_de_v2.clj

```diff
@@ -59,6 +60,31 @@
   [agave app-id]
   (mapv :id (:inputs (.getApp agave app-id))))
 
+(defn list-app-permissions
+  [agave app-ids]
+  (map #(apps/format-app-permissions % (.listAppPermissions agave %)) app-ids))
+
+(defn has-app-permission
+  [agave username app-id required-level]
+  (try+
+    (let [{:keys [write read execute]} (:permission (.getAppPermission agave app-id username))]
+      (case required-level
+        "own"     write
+        "write"   write
+        "read"    read
+        "execute" execute
+        false))
```

Comment:
This is one case where Agave's permission model doesn't quite match ours. We have permission levels, where `read` always implies `execute`, `write` always implies `read`, etc.. If I'm not mistaken (and it's been a while since I've looked at this, so I very well could be mistaken) then there are probably some edge cases here that could bite us. I would expect those edge cases to be rare, however.